### PR TITLE
Migrated to v1beta1 and added agent based charts

### PR DESCRIPTION
**Pull request by jelkosz**

### CODE REVIEW

File: hypershift-agent-template/templates/hostedcluster.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    type: Agent
+    agent:
+      agentNamespace: {{ .Values.agentNamespace }}
+  infraID: {{ .Release.Name }}
+  dns:
+    baseDomain: {{ .Values.baseDnsDomain }}
+  services:
+  - service: APIServer
+    servicePublishingStrategy:
+      type: LoadBalancer
```
Comment (machacekondra) [inline hypershift-agent-template/templates/hostedcluster.yaml:30]: So let remove the nodeport support?

Comment (rawagner) [reply hypershift-agent-template/templates/hostedcluster.yaml:30]: I guess this is just an oversight. It should be

```yaml
- service: APIServer
   servicePublishingStrategy:
     type: {{ .Values.APIPublishingStrategy }}
     {{- if .Values.NodePortAddress  }}
     nodePort:
       address: {{ .Values.NodePortAddress }}
     {{- end }}
```

Comment (jelkosz) [reply hypershift-agent-template/templates/hostedcluster.yaml:30]: indeed, fixed

**Pull request merged**